internal/subscriber/model.go

```diff
@@ -151,14 +152,29 @@ func (a *AckMessage) HasHitDeadline() bool {
 }
 
 // ParseAckID ...
-func ParseAckID(ackID string) *AckMessage {
+func ParseAckID(ackID string) (*AckMessage, error) {
 	// split the message and parse tokens
 	parts := strings.Split(ackID, ackIDSeparator)
+	if !isValidAckID(parts) {
+		return nil, merror.Newf(merror.InvalidArgument, "AckID received is not in expected format")
```

Comment:
store the error in a var and return that.

Reply:
Log the error at the origin. This way the caller need not explicitly log your errors everytime

Reply:
So this function didn't have any context. So I have changed the signature and taking in context from caller method and logging the error in this function itself before passing it back to the caller. Do post any suggestions/changes in this approach